### PR TITLE
Use new version of jtreg.

### PR DESCRIPTION
**Pull request by smillst**
This is the download link found at http://openjdk.java.net/jtreg/

(I want Travis to run successfully before merging.)

### CODE REVIEW

File: checker/build.xml [hunk 1]
```diff
@@ -982,7 +982,7 @@
 
     <target name="-jtreg.download" depends="jtreg.check" unless="jtreg.exists">
       <get
-        src="https://adopt-openjdk.ci.cloudbees.com/view/OpenJDK/job/jtreg/lastSuccessfulBuild/artifact/jtreg-4.1-b12.tar.gz"
+        src="https://adopt-openjdk.ci.cloudbees.com/job/jtreg/lastSuccessfulBuild/artifact/jtreg-4.2-b02.tar.gz"
```
Comment (wmdietl) [inline checker/build.xml]: What do you think of using `https://adopt-openjdk.ci.cloudbees.com/job/jtreg/lastSuccessfulBuild/artifact/jtreg-4.2.0-tip.tar.gz` to automatically get updates within 4.2.0 automatically? jtreg in general seems quite stable.

**Comment by wmdietl** [comment]
I restarted the travis build to see whether the failure is temporary.

**Comment by smillst** [comment]
Still failing, I restarted one more time.

**Comment by smillst** [comment]
Seems like the newer version isn't compatible with [our tests](https://travis-ci.org/typetools/checker-framework/builds/127880666):
````
BUILD FAILED
/home/travis/build/typetools/checker-framework/checker/build.xml:1000: The following error occurred while executing this line:
/home/travis/build/typetools/checker-framework/checker/build.xml:1069: java.lang.NullPointerException
	at com.sun.javatest.regtest.RegressionReporter.fixupReportFiles(RegressionReporter.java:264)
	at com.sun.javatest.regtest.RegressionReporter.fixupReports(RegressionReporter.java:248)
	at com.sun.javatest.regtest.RegressionReporter.report(RegressionReporter.java:107)
	at com.sun.javatest.regtest.Main.batchHarness(Main.java:1943)
	at com.sun.javatest.regtest.Main.run(Main.java:1304)
	at com.sun.javatest.regtest.Main.access$4900(Main.java:99)
	at com.sun.javatest.regtest.Main$Ant.execute(Main.java:905)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:348)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:68)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:348)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:398)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:348)
	at org.apache.tools.ant.Target.execute(Target.java:390)
	at org.apache.tools.ant.Target.performTasks(Target.java:411)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1399)
	at org.apache.tools.ant.Project.executeTarget(Project.java:1368)
	at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1251)
	at org.apache.tools.ant.Main.runBuild(Main.java:809)
	at org.apache.tools.ant.Main.startAnt(Main.java:217)
	at org.apache.tools.ant.launch.Launcher.run(Launcher.java:280)
	at org.apache.tools.ant.launch.Launcher.main(Launcher.java:109)
````

**Comment by smillst** [comment]
Rebased version of these changes pushed in 7e9b4dd.

**Pull request closed without merging**